src/tests/ftest/util/soak_utils.py

```diff
@@ -1288,52 +1291,59 @@ def create_app_cmdline(self, job_spec, pool, ppn, nodesperjob):
-            job_spec, file_oclass, nodesperjob * ppn, nodesperjob, ppn)
-        # include dfuse cmdlines
-        if posix:
-            dfuse, dfuse_start_cmdlist = start_dfuse(
-                self, pool, self.container[-1], name=log_name, job_spec=job_spec)
-            sbatch_cmds.extend(dfuse_start_cmdlist)
-        # allow apps that use an mpi other than default (self.mpi_module)
-        if mpi_module != self.mpi_module:
-            sbatch_cmds.append("module load {}".format(mpi_module))
-        mpirun_cmd = Mpirun(app_cmd, False, mpi_module)
-        mpirun_cmd.get_params(self)
-        if "mpich" in mpi_module:
-            # Pass pool and container information to the commands
-            env = EnvironmentVariables()
-            env["DAOS_UNS_PREFIX"] = format_path(pool, self.container[-1])
-            env["D_LOG_FILE_APPEND_PID"] = "1"
-            mpirun_cmd.assign_environment(env, True)
-        mpirun_cmd.assign_processes(nodesperjob * ppn)
-        mpirun_cmd.ppn.update(ppn)
-        if posix:
-            mpirun_cmd.working_dir.update(dfuse.mount_dir.value)
-        cmdline = "{}".format(str(mpirun_cmd))
-        sbatch_cmds.append(str(cmdline))
-        sbatch_cmds.append("status=$?")
-        if posix:
+        for api in api_list:
+            if not self.enable_il and api in ["POSIX-LIBIOIL", "POSIX-LIBPIL4DFS"]:
+                continue
+            add_containers(self, pool, file_oclass, dir_oclass)
+            sbatch_cmds = ["module purge", "module load {}".format(self.mpi_module)]
+            log_name = "{}_{}_{}_{}_{}_{}".format(
+                job_spec, api, file_oclass, nodesperjob * ppn, nodesperjob, ppn)
+            # include dfuse cmdlines
+            if api in ["POSIX", "POSIX-LIBIOIL", "POSIX-LIBPIL4DFS"]:
+                dfuse, dfuse_start_cmdlist = start_dfuse(
+                    self, pool, self.container[-1], name=log_name, job_spec=job_spec)
+                sbatch_cmds.extend(dfuse_start_cmdlist)
+            # allow apps that use an mpi other than default (self.mpi_module)
             if mpi_module != self.mpi_module:
-                sbatch_cmds.extend(["module purge", "module load {}".format(self.mpi_module)])
-            sbatch_cmds.extend(stop_dfuse(dfuse))
-        commands.append([sbatch_cmds, log_name])
-        self.log.info("<<{} cmdlines>>:".format(job_spec.upper()))
-        for cmd in sbatch_cmds:
-            self.log.info("%s", cmd)
-        if mpi_module != self.mpi_module:
-            mpirun_cmd = Mpirun(app_cmd, False, self.mpi_module)
+                sbatch_cmds.append("module load {}".format(mpi_module))
+            mpirun_cmd = Mpirun(app_cmd, False, mpi_module)
             mpirun_cmd.get_params(self)
+            if "mpich" in mpi_module:
+                # Pass pool and container information to the commands
+                env = EnvironmentVariables()
+                env["DAOS_UNS_PREFIX"] = format_path(pool, self.container[-1])
```

Comment:
Curious, do you know what all still uses `DAOS_UNS_PREFIX`? Do we still NEED this?

Reply:
The ior_utils still uses this env if mpirun is mpich so I have still included this. But I will verify with Mohamad

Reply:
Thanks. It would be great if we can remove this entirely from ftest. (outside the scope of this PR)

Reply:
Related, ior/mdtest in ftest also use `IOR_HINT__MPI__romio_daos_obj_class`.
And mdtest exports `DAOS_POOL` and `DAOS_CONT`

Reply:
soak does get_default_env to define those for ior and mdtest.